pandas/io/excel.py

```diff
@@ -634,14 +634,17 @@ def _parse_cell(cell_contents, cell_typ):
                 else:
                     offset = 1 + max(header)
 
-                for col in index_col:
-                    last = data[offset][col]
-
-                    for row in range(offset + 1, len(data)):
-                        if data[row][col] == '' or data[row][col] is None:
-                            data[row][col] = last
-                        else:
-                            last = data[row][col]
+                # Check if we have an empty dataset
+                # before trying to collect data.
+                if offset < len(data):
```

Comment:
Do we have a test where `offset` is equal to `len(data)`? Just want to make sure we're not off by one, and it's not clear to be that we're not (without having read the code).

Reply:
> What are the changes to the excel file?

To add a test that addresses your question:

> Do we have a test where offset is equal to len(data)?

Don't worry, I accounted for the off-by-one 😉